### DISCUSSION TITLE
Seperate databases vs. everythig in one databases

### DISCUSSION THREAD

**Discussion by cultur98** [oetiker/rrdtool-1.x#1210] [Q&A] [answered]
This is rather a question then a feature request:

I am redesigning my software for environmental 
monitoring.
The RRD is valuable component in my system.
Now I have the idea of using individual databases
for each and every DS I want to store.
Is this a disadvantage instead of putting everything in one database?
I was thinking that the number of write requests within the database should be the same, should'n it?

Thanks in advance for any advise!

**Comment by oetiker** [comment, marked as answer]
Having multiple DS in a single RRD file is much more performant than having a single RRD file for each DS. Modern computers will write 4k Byte every time they write to disk. Regardless of how much data you are changing. So as long as your updates to the RRD file will be less than 4k, it will not make a difference if you write a single or 500 DS to the file.

That said, having a single RRD file for each DS makes you much more flexible in adding and removing DS as you go along. So if performance is not the issue, I would go for the one DS one RRD ... or maybe some hybrid ... like if you have 'in/out' type data these two could go into a single RRD file.

HTH
Tobi

**Reply by cultur98** [reply]
Thanks for this quick and helpful response Tobi!
Best regards,
Hartmut

**Comment by TheWitness** [comment]
@oetiker, I'm just implementing ptiles in DSstats, and I was amazed at how "performant" RRDtool is.  I will be eternally be grateful for that.

To the argument around one data source per RRDfile, this is a troupe put out when graphite became a thing.  In this case, I defer to @oetiker, and I agree with his point.

I could go on with a point by point defense of it but why? I'll be 61 in March.  Not got the time for that now. 

God bless everyone!
